### DISCUSSION TITLE
Avalonia in WinPE supportability

### DISCUSSION THREAD

**Discussion by nixtar** [AvaloniaUI/Avalonia#8011] [Q&A]
Hey all,

At my work we have a closed source LOB WinForms app that runs from [WinPE](https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/winpe-intro).

I want to rewrite it in something more modern as the current codebase is pure spaghetti.

WPF was my first option but after learning that its technically not supported in WinPE ( but appears to run ), I wanted to give Avalonia a chance.
We are a Microsoft shop with all the expensive support contracts so we normally try and keep things "supported", but if we are going outside of support with WPF... Why not use Avalonia?

I did a quick PoC I did using the Control Catalog project + self contained dotnet 6 and it appears to run without any obvious issues.
No hardware rendering but that's to be expected in PE.

Our biggest concern is that we will hit a roadblock after considerable effort has already been spent or an update to Avalonia/WinPE breaks something.
I wanted to post this to see if I could sniff out the "vibes" around what the support would be for WinPE in this scenario and also see if anyone knows of any successful deployments of Avalonia in WinPE?

Cheers

**Comment by robloo** [comment]
You might want to reach out to the Avalonia team directly in private at [team@avaloniaui.net](mailto:team@avaloniaui.net)

https://github.com/AvaloniaUI/Avalonia/issues/4065

If you are willing to pay for support I'm sure they will commit to quickly fixing any issues you run across in production. Since Avalonia already runs just fine on WinPE according to your own testing, it seems support just to make sure things stay stable is most important. I doubt there are any fundamental issues with your plan but I'm sure the team will let you know of any concerns.

**Reply by nixtar** [reply]
Thanks for the tip. I didn't know they had a team mailbox 👍.

At this stage I was hoping to also find someone in the community with some experience in this space.

I'll wait a few days then mark this as answered.

If I hear anything directly from the team and they are happy with me sharing the details I'll update this post.

From Avalonias prospective I could see it perfectly reasonable to just say "WinPE is not on this list https://docs.avaloniaui.net/#supported-platforms sooo cya later". Would a customer paying for Avalonia support get support for Win7?

**Comment by ArchibaldMacdonald-cyacomb** [comment]
Recently ran a sample Avalonia (11.0.5) application on WinPE and it does not seem to startup successfully, I am getting an exit code 255. @nixtar if you are still using Avalonia/WinPE what was the version you were using successfully?

**Reply by timunie** [reply]
We have a PR for adding back WinPE compatibility. Please give it a try: #14090. You can test it using nightly builds or wait for 11.1 to drop.

**Reply by nixtar** [reply]
Around the time I posted this back in 2022 I did a very basic PoC as well as ran the control sample app for whatever was the latest stable avalonia 0.10 back then. Never tried v11, the project in question is on hold for now.

Awesome to see its being looked at for 11.1.
Our plans was to make sure avalonia would commit to supporting winpe as a platform and purchase support if/when we greenlight the project.

@timunie if I get a chance I'll have a poke at with some of the various winpe versions we use. ❤️

**Reply by ArchibaldMacdonald-cyacomb** [reply]
Fantastic, thanks @timunie  I'll give it a whirl and get back to you.

**Reply by timunie** [reply]
> Our plans was to make sure avalonia would commit to supporting winpe as a platform and purchase support if/when we greenlight the project. 

reach out to the team via mail if you have questions about paid support: https://avaloniaui.net/Support

**Reply by ArchibaldMacdonald-cyacomb** [reply]
Hi @timunie I can confirm that the sample `AppWithoutLifetime` (commit: dac4ae422da17afb717c8e60e0728a1d8b1b4751) works with the ADK version 10.1.22621.1.

~A project of ours built against the Nightly is not currently working, but this may well be unrelated to Avalonia so I will continue to investigate why.~ (Edit: was a Problem on my end. Avalonia is working flawlessly.)

Thanks for your help.

Cheers,
Archie
